src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/CosmosDBSourceTask.java

```diff
@@ -90,8 +91,17 @@ public void start(Map<String, String> map) {
                 if(node == null) {
                     continue;
                 }
+
+                // Set the Kafka message key if option is provided and "id" field is set in document
+                String messageKey = "";
+                if (node.get("id") != null && this.settings.isSetMessageKey()) {
+                    messageKey = node.get("id").toString();
+                }
+
                 // Since Lease container takes care of maintaining state we don't have to send source offset to kafka
-                SourceRecord sourceRecord = new SourceRecord(partition, null, topic, null, node.toString());
+                SourceRecord sourceRecord = new SourceRecord(partition, null, topic,
+                                                    Schema.STRING_SCHEMA, messageKey, 
+                                                    Schema.STRING_SCHEMA, node.toString());
```

Comment:
should we be toString() here? does SourceRecord expect a string? or can it take in node?

Reply:
I don't think it can cause that would require a schema object along with the JSON value.

Reply:
at some point we will need to think about how we handle this, when it comes to handling other converters. i guess the converter will just have to know how to take a json string and covert it to avro, or other.